RecoMuon/MuonIdentification/src/MuonTimingFiller.cc

```diff
@@ -44,11 +45,21 @@ MuonTimingFiller::MuonTimingFiller(const edm::ParameterSet& iConfig, edm::Consum
 {
    // Load parameters for the DTTimingExtractor
    edm::ParameterSet dtTimingParameters = iConfig.getParameter<edm::ParameterSet>("DTTimingParameters");
-   theDTTimingExtractor_ = new DTTimingExtractor(dtTimingParameters,iC);
 
    // Load parameters for the CSCTimingExtractor
    edm::ParameterSet cscTimingParameters = iConfig.getParameter<edm::ParameterSet>("CSCTimingParameters");
-   theCSCTimingExtractor_ = new CSCTimingExtractor(cscTimingParameters,iC);
+
+   // Fallback mechanism for old configs (there the segment matcher was built inside the timing extractors)
+   edm::ParameterSet matchParameters;
+   if (iConfig.existsAs<edm::ParameterSet>("MatchParameters")) 
+       matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
+     else 
+       matchParameters = dtTimingParameters.getParameter<edm::ParameterSet>("MatchParameters");
+   
+   
+   theMatcher_ = new MuonSegmentMatcher(matchParameters, iC);
```

Comment:
On Thu, Jan 21, 2016 at 7:00 AM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoMuon/MuonIdentification/src/MuonTimingFiller.cc
> <https://github.com/cms-sw/cmssw/pull/13010#discussion_r50363402>:
>
> >
> >     // Load parameters for the CSCTimingExtractor
> >     edm::ParameterSet cscTimingParameters = iConfig.getParameter<edm::ParameterSet>("CSCTimingParameters");
> > -   theCSCTimingExtractor_ = new CSCTimingExtractor(cscTimingParameters,iC);
> > +
> > +   // Fallback mechanism for old configs (there the segment matcher was built inside the timing extractors)
> > +   edm::ParameterSet matchParameters;
> > +   if (iConfig.existsAs<edm::ParameterSet>("MatchParameters"))
> > +       matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
> > +     else
> > +       matchParameters = dtTimingParameters.getParameter<edm::ParameterSet>("MatchParameters");
> > +
> > +
> > +   theMatcher_ = new MuonSegmentMatcher(matchParameters, iC);
>
> it could be a good opportunity to switch to unique_ptr to let it handle
> the lifetime of the object and drop the delete
>
I'll do that, tomorrow if I find the time. I guess I can make another PR
for that once this one goes in.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13010/files#r50363402>.
>